src/main/scala/sbtdatabricks/DatabricksPlugin.scala

```diff
@@ -64,29 +64,51 @@ object DatabricksPlugin extends AutoPlugin {
       }
     }.flatMap(c => c)
   }
-  
+
+  /** Returns all the jars related to this library. */
   private lazy val dbcClasspath = Def.task {
-    ((Keys.`package` in Compile).value +: (managedClasspath in Runtime).value.files
+    (dbcLocalProjects.value ++ (managedClasspath in Runtime).value.files
       ).filterNot(_.getName startsWith "scala-")
   }
 
+  /** Walks the dependency tree of local projects and packages them. */
+  private def dbcLocalProjects: Def.Initialize[Task[Seq[File]]] =
+    (thisProjectRef, thisProject, state).flatMap {
+      (projectRef: ProjectRef, project: ResolvedProject, currentState: State) => {
+        def visit(p: ProjectRef): Seq[Task[java.io.File]] = {
+          val extracted = Project.extract(currentState)
+          val data = extracted.structure.data
+          val depProject = thisProject in p get data getOrElse sys.error("Invalid project: " + p)
+          val jarFile = (Keys.`package` in (p, ConfigKey("runtime"))).get(data).get
+          jarFile +: depProject.dependencies.map {
+            case ResolvedClasspathDependency(dep, confMapping) => dep
+          }.flatMap(visit).toList
```

Comment:
Please leave a comment explaining what's happening here, are dependencies themselves `ProjectRef`s that we can recursively visit?